File: solutions/0467-unique-substrings-in-wraparound-string/solution.cpp
class Solution {
public:
    bool check(char b,char a){
        if(b - a == 1){
            return true;
        }
        else if(b == 'a' && a == 'z'){
            return true;
        }
        return false;
    }
    int findSubstringInWraproundString(string s) {
        int n = s.length();
        vector<int> v(n,1);
        for(int i = 1;i<s.length();i++){
            if(check(s[i],s[i-1])){
                v[i] = v[i-1]+1;
            }
        }
        vector<int> ans(26,0);
        for(int i = 0;i<s.length();i++){
            ans[s[i]-'a'] = max(ans[s[i]-'a'],v[i]);
        }
        int sum = 0;
        for(auto it:ans){
            sum+=it;
        }
        return sum;
    }
};

File: solutions/0467-unique-substrings-in-wraparound-string/solution.py
class Solution:
    def findSubstringInWraproundString(self, p: str) -> int:
        cnt = 0
        seen = {} # longest string ending at char 
        for i in range(len(p)): 
            if 0 < i and (ord(p[i]) - ord(p[i-1])) % 26 != 1: cnt = 0 # reset counter 
            cnt += 1
            seen[p[i]] = max(seen.get(p[i], 0), cnt)
        return sum(seen.values())
